pkg/operation/botanist/component/networkpolicies/peers.go

```diff
@@ -98,8 +98,10 @@ func excludeBlock(parentBlock *net.IPNet, cidrs ...string) ([]string, error) {
 		if err != nil {
 			return matchedCIDRs, err
 		}
+		parentBlockMaskLength, _ := parentBlock.Mask.Size()
+		ipNetMaskLength, _ := ipNet.Mask.Size()
 
-		if parentBlock.Contains(ip) && !ipNet.Contains(parentBlock.IP) {
+		if parentBlock.Contains(ip) && (parentBlockMaskLength < ipNetMaskLength) {
```

Comment:
With this change we will exclude all blocks if they are in the parent block AND smaller.
What about equal size blocks? Should they be excluded?
If the parent block is indeed smaller what should happen then?